### ISSUE TITLE
Does not inherit from prog-mode

### ISSUE THREAD

**Issue by justmytwospence** [emacs-ess/ESS#219]
Is there a particular reason ESS does not inherit from prog-mode? See: https://www.gnu.org/software/emacs/manual/html_node/elisp/Basic-Major-Modes.html.  All programming modes are "supposed" to inherit from this, so that they can be configured together with prog-mode-hook.

**Comment by vspinu** [comment]
My guess is that ESS was built when prog-mode wasn't there as yet. This
should be fixed with high priority.

**Comment by mmaechler** [comment]
maybe re-considering this _high priority_ fix  now would be a good time.

**Comment by LaDilettante** [comment]
I would highly second this, as I tend to enable `company-mode` using `prog-mode-hook`, and ESS is not included unfortunately.

**Comment by fikovnik** [comment]
Yes! Please :-)

**Comment by saidie** [comment]
👍

**Comment by jabranham** [comment]
While 96e1e5b may have made ESS literally inherit from prog-mode, it doesn't quite solve the problem. For example, prog-mode-hook does not get run in ess modes still.

**Comment by lionel-** [comment]
Good point, we'll do it manually. One day we'll have properly derived modes...
